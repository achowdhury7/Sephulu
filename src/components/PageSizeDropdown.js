import React, { Component } from 'react'
import { DropdownButton, MenuItem } from 'react-bootstrap'

class PageSizeDropdown extends Component {
  constructor(props) {
    super(props)

    this.state = {
      activeKey: '25'
    }

    this.handleSelect = this.handleSelect.bind(this)
  }

  handleSelect(key) {
    if (this.state.activeKey != key) {
      this.setState({
        activeKey: key
      })
  
      if (this.props.handleSelect) this.props.handleSelect(key)
    }
  }

  render() {
    const menuItems = this.props.menuItems.map(item => (
      <MenuItem key={item} eventKey={item}>{item}</MenuItem>
    ))

    return (
      <DropdownButton
        id="page-size"
        title={this.state.activeKey}
        onSelect={this.handleSelect}
      >
        {menuItems}
      </DropdownButton>
    )
  }
}

export default PageSizeDropdown
