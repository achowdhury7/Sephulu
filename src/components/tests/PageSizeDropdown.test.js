import React from 'react'
import { shallow, mount, render } from 'enzyme'

import PageSizeDropdown from '../PageSizeDropdown'

test('render a dropdown', () => {
  const menuItems = ['a', 'b', 'c']
  const wrapper = shallow(
    <PageSizeDropdown menuItems={menuItems} />
  )
  expect(wrapper).toMatchSnapshot()
})

test('set `activeKey` on menu item click', () => {
  const onSelect = jest.fn()
  const menuItems = ['a', 'b', 'c']
  const wrapper = mount(
    <PageSizeDropdown menuItems={menuItems} onSelect={onSelect} />
  )

  expect(wrapper).toMatchSnapshot()

  wrapper.find('a').first().simulate('click')
  expect(wrapper.state('activeKey')).toEqual('a')
})
