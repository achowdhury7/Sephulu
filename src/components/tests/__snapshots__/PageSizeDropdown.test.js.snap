// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render a dropdown 1`] = `
<DropdownButton
  id="page-size"
  onSelect={[Function]}
  title="25"
>
  <MenuItem
    bsClass="dropdown"
    disabled={false}
    divider={false}
    eventKey="a"
    header={false}
    id="a"
    key="a"
  >
    a
  </MenuItem>
  <MenuItem
    bsClass="dropdown"
    disabled={false}
    divider={false}
    eventKey="b"
    header={false}
    id="b"
    key="b"
  >
    b
  </MenuItem>
  <MenuItem
    bsClass="dropdown"
    disabled={false}
    divider={false}
    eventKey="c"
    header={false}
    id="c"
    key="c"
  >
    c
  </MenuItem>
</DropdownButton>
`;

exports[`set \`activeKey\` on menu item click 1`] = `
<PageSizeDropdown
  menuItems={
    Array [
      "a",
      "b",
      "c",
    ]
  }
  onSelect={[MockFunction]}
>
  <DropdownButton
    id="page-size"
    onSelect={[Function]}
    title="25"
  >
    <Uncontrolled(Dropdown)
      id="page-size"
      onSelect={[Function]}
    >
      <Dropdown
        bsClass="dropdown"
        componentClass={[Function]}
        id="page-size"
        onSelect={[Function]}
        onToggle={[Function]}
      >
        <ButtonGroup
          block={false}
          bsClass="btn-group"
          className="dropdown"
          justified={false}
          vertical={false}
        >
          <div
            className="dropdown btn-group"
          >
            <DropdownToggle
              bsClass="dropdown-toggle"
              bsRole="toggle"
              id="page-size"
              key=".0"
              onClick={[Function]}
              onKeyDown={[Function]}
              open={false}
              useAnchor={false}
            >
              <Button
                active={false}
                aria-expanded={false}
                aria-haspopup={true}
                block={false}
                bsClass="btn"
                bsStyle="default"
                className="dropdown-toggle"
                disabled={false}
                id="page-size"
                onClick={[Function]}
                onKeyDown={[Function]}
                role="button"
              >
                <button
                  aria-expanded={false}
                  aria-haspopup={true}
                  className="dropdown-toggle btn btn-default"
                  disabled={false}
                  id="page-size"
                  onClick={[Function]}
                  onKeyDown={[Function]}
                  role="button"
                  type="button"
                >
                  25
                   
                  <span
                    className="caret"
                  />
                </button>
              </Button>
            </DropdownToggle>
            <DropdownMenu
              bsClass="dropdown-menu"
              bsRole="menu"
              key=".1"
              labelledBy="page-size"
              onClose={[Function]}
              onSelect={[Function]}
              pullRight={false}
            >
              <RootCloseWrapper
                disabled={true}
                event="click"
                onRootClose={[Function]}
              >
                <ul
                  aria-labelledby="page-size"
                  className="dropdown-menu"
                  role="menu"
                >
                  <MenuItem
                    bsClass="dropdown"
                    disabled={false}
                    divider={false}
                    eventKey="a"
                    header={false}
                    id="a"
                    key=".$a"
                    onKeyDown={[Function]}
                    onSelect={[Function]}
                  >
                    <li
                      className=""
                      role="presentation"
                    >
                      <SafeAnchor
                        componentClass="a"
                        id="a"
                        onClick={[Function]}
                        onKeyDown={[Function]}
                        role="menuitem"
                        tabIndex="-1"
                      >
                        <a
                          href="#"
                          id="a"
                          onClick={[Function]}
                          onKeyDown={[Function]}
                          role="menuitem"
                          tabIndex="-1"
                        >
                          a
                        </a>
                      </SafeAnchor>
                    </li>
                  </MenuItem>
                  <MenuItem
                    bsClass="dropdown"
                    disabled={false}
                    divider={false}
                    eventKey="b"
                    header={false}
                    id="b"
                    key=".$b"
                    onKeyDown={[Function]}
                    onSelect={[Function]}
                  >
                    <li
                      className=""
                      role="presentation"
                    >
                      <SafeAnchor
                        componentClass="a"
                        id="b"
                        onClick={[Function]}
                        onKeyDown={[Function]}
                        role="menuitem"
                        tabIndex="-1"
                      >
                        <a
                          href="#"
                          id="b"
                          onClick={[Function]}
                          onKeyDown={[Function]}
                          role="menuitem"
                          tabIndex="-1"
                        >
                          b
                        </a>
                      </SafeAnchor>
                    </li>
                  </MenuItem>
                  <MenuItem
                    bsClass="dropdown"
                    disabled={false}
                    divider={false}
                    eventKey="c"
                    header={false}
                    id="c"
                    key=".$c"
                    onKeyDown={[Function]}
                    onSelect={[Function]}
                  >
                    <li
                      className=""
                      role="presentation"
                    >
                      <SafeAnchor
                        componentClass="a"
                        id="c"
                        onClick={[Function]}
                        onKeyDown={[Function]}
                        role="menuitem"
                        tabIndex="-1"
                      >
                        <a
                          href="#"
                          id="c"
                          onClick={[Function]}
                          onKeyDown={[Function]}
                          role="menuitem"
                          tabIndex="-1"
                        >
                          c
                        </a>
                      </SafeAnchor>
                    </li>
                  </MenuItem>
                </ul>
              </RootCloseWrapper>
            </DropdownMenu>
          </div>
        </ButtonGroup>
      </Dropdown>
    </Uncontrolled(Dropdown)>
  </DropdownButton>
</PageSizeDropdown>
`;
